### PR TITLE
Add auth_type_id and update rcode_id

### PR DESCRIPTION
**Pull request by Aniak5**
image

### CODE REVIEW

**Comment by Aniak5** [comment]
The changes to `rcode_id` are so that the soon to come LDAP class can use this attribute.

**Review by rroupski — APPROVED** [review]

File: events/network/rfb.json [hunk 1]
```diff
@@ -6,6 +6,49 @@
   "uid": 9,
   "attributes": {
     "auth_type": {
+      "requirement": "optional"
+    },
+    "auth_type_id": {
+      "description": "The normalized identifier of the agreed upon RFB authentication type. See <a target='_blank' href='https://www.rfc-editor.org/rfc/rfc6143.html'>RFC-6143</a>.",
+      "enum": {
```
Comment (floydtree) [inline events/network/rfb.json:13]: 99 instead of -1?

File: events/network/dns.json [hunk 2]
```diff
@@ -29,10 +29,107 @@
...
     "rcode": {
+      "description": "The DNS server response code, as defined by the event source.",
       "group": "primary",
       "requirement": "optional"
     },
     "rcode_id": {
+      "caption": "Response Code ID",
+      "description": "The normalized identifier of the DNS server response code. See <a target='_blank' href='https://datatracker.ietf.org/doc/html/rfc6895'>RFC-6895</a>.",
+      "enum": {
+        "-1": {
```
Comment (floydtree) [inline events/network/dns.json:40]: 99 instead of -1

**Review by floydtree — CHANGES_REQUESTED** [review]

**Review by mr-splunk — APPROVED** [review]

File: dictionary.json [hunk 3]
```diff
@@ -246,9 +246,23 @@
       "description": "The agreed upon authentication type.",
       "type": "string_t"
     },
+    "auth_type_id": {
+      "caption": "Authentication Type ID",
+      "description": "The normalized identifier of the agreed upon authentication type. See specific usage.",
+      "enum": {
+        "-1": {
```
Comment (floydtree) [inline dictionary.json:253]: 99 instead of -1

**Review by floydtree — CHANGES_REQUESTED** [review]

File: dictionary.json (hunk 3 above)
Comment (Aniak5) [reply dictionary.json]: Oops, missed that one. Done, thanks!

**Review by floydtree — APPROVED** [review]

**Pull request merged**